scripts/upgrade.sh

```diff
@@ -0,0 +1,190 @@
+#!/bin/bash
+
+set -e
+set -u
+set -o pipefail
+
+__ScriptName="$(basename "$0")"
+SALT_CALL=${SALT_CALL:-salt-call}
+
+VERBOSE=${VERBOSE:-0}
+LOGFILE=/var/log/metalk8s-upgrade.log
+DESTINATION_VERSION=""
+DRY_RUN=0
+
+# Some helpful guide on upgrade options
+__usage() {
+    cat << EOF
+  Usage :  ${__ScriptName} -t <destination-version> [options]
+  Options:
+    -h | --help                  Display HELP options
+    -d | --dry-run               Show Pre-Check output.
+         --destination-version   Upgrade to specified version
+    -l | --log-file              Path to log file
+    -v | --verbose               Run in verbose mode
+EOF
+exit
+}
+
+# # handle the default case if script is called with no args
+[ "$#" -eq 0 ] && __usage
+
+if ! options=$(getopt --options dvhl: --long dry-run,verbose,help,destination-version:,log-file: -- "$@"); then
+    echo 1>&2 "Incorrect arguments: Provide a valid MetalK8s version"
+    exit 1
+fi
+eval set -- "$options"
+
+while true; do
+    case "$1" in
+        -d|--dry-run)
+            DRY_RUN=1
+            shift;;
+        -v|--verbose)
+            VERBOSE=1;
+            shift;;
+        --destination-version)
+            DESTINATION_VERSION="$2"
+            shift 2;;
+        -l|--log-file)
+            LOGFILE="$2"
+            shift 2;;
+        -h|--help)
+            __usage
+            shift;;
+        --)
+            shift;
+            break;;
+        *) # unhandled inputs
+            echo "K.O   Incorrect arguments" 1>&2
+            exit 1 ;;
+    esac
+done
```

Comment:
```
while (( "$#" )); do
  case "$1" in
    -d|--dry-run)
      DRY_RUN=1
      shift
      ;;
    -v|--verbose)
      VERBOSE=1
      shift
      ;;
    --destination-version)
      DESTINATION_VERSION="$2"
      shift 2
      ;;
    -l|--log-file)
      LOGFILE="$2"
      shift 2
      ;;
    -h|--help)
      _usage
      exit
      ;;
    *) # unsupported flags
      echo "Error: Unsupported flag $1" >&2
      _usage
      exit 1
      ;;
  esac
done
```
Please really stick to the same script as downgrade one so that it will be easier to maintain